### PR TITLE
Java formatting based on Google Java Style

### PR DESCRIPTION
**Pull request by octonato**
This is adding java formatter using github.com/diffplug/spotless (best I found so far)

The main goal is to format using 2 spaces for tabs fo we have less horizontal scrolling for java sources. 

The best way to review is to generate the docs and navigate through it. 

Draft because on top of #338.

Refs: #272 and #318

### CODE REVIEW

**Review by patriknw — COMMENTED** [review]
good

File: docs-source/docs/modules/microservices-tutorial/pages/complete-entity.adoc [hunk 1]
```diff
@@ -302,7 +308,7 @@ Scala::
 include::example$03-shopping-cart-service-scala/src/main/scala/shopping/cart/ShoppingCartServiceImpl.scala[tag=checkoutAndGet]
 ----
 
-ifdef::todo[TODO: refer to the source download, especially for the optional commands that are not shown here]
+As mentioned at the beginning of this page, `RemoveItem` and `AdjustItemQuantity` commands are not mandatory for the subsequent steps of the tutorial. A zip file with the completed for this page implementing the optional commands is available for [.group-scala]#link:_attachments/3-shopping-cart-event-sourced-complete-scala.zip[download]# [.group-java]#link:_attachments/3-shopping-cart-event-sourced-complete-java.zip[download]#.
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/complete-entity.adoc:311]:
```suggestion
As mentioned at the beginning of this page, `RemoveItem` and `AdjustItemQuantity` commands are not mandatory for the subsequent steps of the tutorial. A zip file with the completed code for this page implementing the optional commands is available for [.group-scala]#link:_attachments/3-shopping-cart-event-sourced-complete-scala.zip[download]# [.group-java]#link:_attachments/3-shopping-cart-event-sourced-complete-java.zip[download]#.
```

File: docs-source/docs/modules/microservices-tutorial/pages/entity.adoc [hunk 2]
```diff
@@ -238,12 +237,19 @@ include::example$02-shopping-cart-service-scala/src/test/scala/shopping/cart/Sho
...
 ----
 
-You can learn more about the `EventSourcedBehaviorTestKit` in the {akka}/typed/persistence-testing.html#unit-testing[Akka reference documentation] {akka}/typed/persistence-testing.html#unit-testing[{tab-icon}, window="tab"]
+[.group-java]
+[source,shell script]
+----
+mvn test
+----
+
+You can learn more about the [.group-scala]#`EventSourcedBehaviorTestKit`# [.group-java]#`TestKitJunitResource`# in the {akka}/typed/persistence-testing.html#unit-testing[Akka reference documentation] {akka}/typed/persistence-testing.html#unit-testing[{tab-icon}, window="tab"]
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/entity.adoc:252]:
```suggestion
You can learn more about the [.group-scala]#`EventSourcedBehaviorTestKit`# [.group-java]#`TestKitJunitResource`# in the {akka}/typed/persistence-testing.html#unit-testing[Akka Persistence reference documentation] {akka}/typed/persistence-testing.html#unit-testing[{tab-icon}, window="tab"]
```
I think @patriknw tried to standardize those links to `Akka <module name> reference documentation`

**Review by ignasi35 — COMMENTED** [review]
Only reviewed the `.adoc` files (not the sources and the use of tags in the sources).
Left two small suggestions.

**Review by ignasi35 — COMMENTED** [review]
Almost all code from `00-shopping-cart-service-java` is gone?

**Comment by ignasi35** [comment]
@octonato did you target the appropriate branch in the PR?

File: docs-source/docs/modules/microservices-tutorial/pages/complete-entity.adoc (hunk 1 above)
Comment (ignasi35) [reply docs-source/docs/modules/microservices-tutorial/pages/complete-entity.adoc:311]: I just noticed this PR is on top of other PRs so this suggestion is probably in the wrong PR.

File: docs-source/docs/modules/microservices-tutorial/pages/entity.adoc (hunk 2 above)
Comment (ignasi35) [reply docs-source/docs/modules/microservices-tutorial/pages/entity.adoc:252]: I just noticed this PR is on top of other PRs so this suggestion is probably in the wrong PR.

**Comment by ignasi35** [comment]
> Almost all code from `00-shopping-cart-service-java` is gone?

Because `00-` needs tha bare minimum and it had the final result. This code removal is OK.

File: docs-source/docs/modules/microservices-tutorial/pages/complete-entity.adoc (hunk 1 above)
Comment (octonato) [reply docs-source/docs/modules/microservices-tutorial/pages/complete-entity.adoc:311]: yes, and I changed that text there as well.

**Comment by octonato** [comment]
> @octonato did you target the appropriate branch in the PR?

@ignasi35, nope.

Fixed now.

**Comment by ihostage** [comment]
> This is adding java formatter using github.com/diffplug/spotless (best I found so far)

@octonato How about Checktyle Plugin with predefined _Google Java Style Guide_ rules? 😄 
https://checkstyle.sourceforge.io/google_style.html

**Comment by octonato** [comment]
@ihostage, any formatter is good, we aren't much opinionated on that, we just want consistency and something that make it look good in the page snippets. 

One problem with the one I chose is that it requires JDK11. Ok for travis, but developers can eventually hit that issue. 

If the one you are suggesting is compiled with JDK8, then is better.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**